crates/sui-sdk/src/json_rpc_error.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use jsonrpsee::types::{error::UNKNOWN_ERROR_CODE, ErrorObjectOwned};
+pub use sui_json_rpc::error::{TRANSACTION_EXECUTION_CLIENT_ERROR_CODE, TRANSIENT_ERROR_CODE};
+use thiserror::Error;
+
+#[derive(Error, Debug, Clone)]
+pub struct Error {
+    pub code: i32,
+    pub message: String,
+    // TODO: as this SDK is specialized for the Sui JSON RPC implementation, we should define structured representation for the data field if applicable
+    pub data: Option<serde_json::Value>,
+}
+
+impl std::fmt::Display for Error {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "code: '{}', message: '{}'", self.code, self.message)
+    }
+}
+
+impl Error {
+    pub fn is_call_error(&self) -> bool {
+        self.code != UNKNOWN_ERROR_CODE
+    }
+
+    pub fn is_client_error(&self) -> bool {
+        use jsonrpsee::types::error::{
```

Comment:
Ditto -- this is a dependency on the crate you're trying to get people to not have to depend on right?